data-raw/workflow/workflow_union_3_years_in_one_table_pseudonymised.sql

```diff
@@ -0,0 +1,232 @@
+              when (len = 6.0 and substr(pcd_temp, 5, 1) = '5') then (substr(pcd_temp, 1, 4) || 's' || substr(pcd_temp, 6, 1))
+              when (len = 6.0 and substr(pcd_temp, 5, 1) = '0') then (substr(pcd_temp, 1, 4) || 'o' || substr(pcd_temp, 6, 1))
+              when (len = 6.0 and substr(pcd_temp, 6, 1) = '5') then (substr(pcd_temp, 1, 5) || 's')
+              when (len = 6.0 and substr(pcd_temp, 6, 1) = '0') then (substr(pcd_temp, 1, 5) || 'o')
+              when (len = 5.0 and substr(pcd_temp, 1, 1) = '5') then ('s' || substr(pcd_temp, 2, 4))
+              when (len = 5.0 and substr(pcd_temp, 1, 1) = '0') then ('o' || substr(pcd_temp, 2, 4))
+              when (len = 5.0 and substr(pcd_temp, 2, 1) = 's') then (substr(pcd_temp, 1, 1) || '5' || substr(pcd_temp, 3, 3))
+              when (len = 5.0 and substr(pcd_temp, 2, 1) = 'o') then (substr(pcd_temp, 1, 1) || '0' || substr(pcd_temp, 3, 3))
+              when (len = 5.0 and substr(pcd_temp, 2, 1) = 'i') then (substr(pcd_temp, 1, 1) || '1' || substr(pcd_temp, 3, 3))
+              when (len = 5.0 and substr(pcd_temp, 2, 1) = 'l') then (substr(pcd_temp, 1, 1) || '1' || substr(pcd_temp, 3, 3))
+              when (len = 5.0 and substr(pcd_temp, 3, 1) = 's') then (substr(pcd_temp, 1, 2) || '5' || substr(pcd_temp, 4, 2))
+              when (len = 5.0 and substr(pcd_temp, 3, 1) = 'o') then (substr(pcd_temp, 1, 2) || '0' || substr(pcd_temp, 4, 2))
+              when (len = 5.0 and substr(pcd_temp, 3, 1) = 'i') then (substr(pcd_temp, 1, 2) || '1' || substr(pcd_temp, 4, 2))
+              when (len = 5.0 and substr(pcd_temp, 3, 1) = 'l') then (substr(pcd_temp, 1, 2) || '1' || substr(pcd_temp, 4, 2))
+              when (len = 5.0 and substr(pcd_temp, 4, 1) = '5') then (substr(pcd_temp, 1, 3) || 's' || substr(pcd_temp, 5, 1))
+              when (len = 5.0 and substr(pcd_temp, 4, 1) = '0') then (substr(pcd_temp, 1, 3) || 'o' || substr(pcd_temp, 5, 1))
+              when (len = 5.0 and substr(pcd_temp, 5, 1) = '5') then (substr(pcd_temp, 1, 4) || 's')
+              when (len = 5.0 and substr(pcd_temp, 5, 1) = '0') then (substr(pcd_temp, 1, 4) || 'o')
+              else pcd_temp
+            end as pcd_temp
+          from (
+            select q01.*, length(postcode) as len, postcode as pcd_temp
+            from (
+              select
+                postcode_old,
+                upper(regexp_replace(postcode, '[^[:alnum:]]', '')) as postcode
+              from (
+                select
+                  postcode_old,
+                  case when (length(postcode) = 0.0) then null when not (length(postcode) = 0.0) then postcode end as postcode
+                from (
+                  select distinct postcode as postcode_old, postcode
+                  from dim.ons_postcode_data_dim
+                  where
+                    (country_code = 'E92000001') and
+                    (not((postcode is null)))
+                ) q01
```

Comment:
Seems strange to call multiple sub-queries the same, `q01`. But that is how dbplyr outputs it so...